packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -266,6 +267,77 @@ module.exports = ({ strapi }) => ({
     return mappedEntity;
   },
 
+  async publishMany(opts, uid) {
+    const params = {
+      ...opts,
+      data: {
+        [PUBLISHED_AT_ATTRIBUTE]: new Date(),
+      },
+    };
+
+    const query = transformParamsToQuery(uid, params);
+    const entitiesToUpdate = await strapi.db.query(uid).findMany(query);
+    // No entities to update, return early
+    if (!entitiesToUpdate.length) {
+      return null;
+    }
```

Comment:
What if one of the entities is already published? Right now we publish it twice.

Reply:
I spotted that too, I was thinking of checking like publish when I run the validation:

```
    await Promise.all(
      entities.map((entityToUpdate) => {
        if (entityToUpdate[PUBLISHED_AT_ATTRIBUTE]) {
          throw new ApplicationError('already.published');
        }

        return strapi.entityValidator.validateEntityCreation(
          strapi.getModel(uid),
          entityToUpdate,
          {
            isDraft: true,
          },
          entityToUpdate
        );
      })
    );
```

Reply:
that sounds good!